src/data/nat/prime.lean

```diff
@@ -645,6 +645,18 @@ have hn : 0 < n := nat.pos_of_ne_zero $ λ h, begin
 end,
 perm_of_prod_eq_prod (by rwa prod_factors hn) h₂ (@prime_of_mem_factors _)
 
+lemma prime_pow_factors (p y : ℕ) [hp : fact p.prime] :
```

Comment:
As I think someone suggested on discord or Zulip, this would likely be more convenient as 
```suggestion
lemma prime.factors_pow {p} (hp : fact p.prime) (y : ℕ) :
```